# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Selection.cpp
# Commit: c1e64e8e4665
# Full Hash: c1e64e8e466506d1d9a36d747df25527bd545b10
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2022-12-01 16:18:29
# Regressor Bug: 1791759
# File Overlap Count: 1
# Description:
#   Bug 1791759 - Add support for `content-visibility: auto` r=emilio
#   
#   This change adds support for `content-visibilty: auto` as well as
#   showing and hiding content based on the relevancy of the content as
#   defined in the specification. Changes to relevancy are handled by
# ==============================================================================

diff -r 6d4d56f57b7b -r c1e64e8e4665 dom/base/Selection.cpp
--- a/dom/base/Selection.cpp	Thu Dec 01 01:23:08 2022 +0200
+++ b/dom/base/Selection.cpp	Wed Nov 30 18:00:27 2022 +0000
@@ -9,6 +9,8 @@
  */
 
 #include "mozilla/dom/Selection.h"
+
+#include "LayoutConstants.h"
 #include "mozilla/intl/BidiEmbeddingLevel.h"
 
 #include "mozilla/AccessibleCaretEventHub.h"
@@ -3200,6 +3202,7 @@
   PresShell* presShell = GetPresShell();
   if (presShell) {
     doc = presShell->GetDocument();
+    presShell->ScheduleContentRelevancyUpdate(ContentRelevancyReason::Selected);
   }
 
   // We've notified all selection listeners even when some of them are removed